### PR TITLE
Improve SignalR Redis Cluster section

### PR DESCRIPTION
**Pull request by VilleKylmamaa**
SignalR Redis backplane documentation informs about the trade-off that increasing the number of nodes in a Redis Cluster decreases the message throughput of the backplane ([source video](https://www.youtube.com/watch?v=6G22a5Iooqk), and [source slides](https://www.slideshare.net/RedisLabs/redis-day-tlv-2018-scaling-redis-pubsub)).

Also mention that in the SignalR app, all the Redis nodes should be included in the Redis connection string or added as `EndPoints` in the `ConfigurationOptions` object.

Reference issue: https://github.com/dotnet/aspnetcore/issues/46408

### CODE REVIEW

**Review by guardrex — CHANGES_REQUESTED** [review]
Thanks @VilleKylmamaa ... ~I'll perform an edit pass after technical review.~ **Done!**

**Comment by VilleKylmamaa** [comment]
I removed the Redis Sentinel suggestion because of uncertainty whether it is supported. You can trigger Sentinel mode in StackExchange.Redis by adding a service name to the connection string (https://stackexchange.github.io/StackExchange.Redis/Configuration), but I am unsure if it works with SignalR. I failed to make it work in quick testing.

**Comment by guardrex** [comment]
@VilleKylmamaa ... See if my updates broke any of the meaning.

**Comment by VilleKylmamaa** [comment]
@guardrex The edits are good 👍

**Comment by guardrex** [comment]
Thanks @VilleKylmamaa ... We'll stand-by for a bit until @BrennanConroy is free to take a look.

File: aspnetcore/signalr/redis-backplane.md [hunk 1]
```diff
@@ -212,9 +212,14 @@ services.AddSignalR()
 
 :::moniker-end
 
-## Redis Clustering
+## Redis Cluster
+
+[Redis Cluster](https://redis.io/topics/cluster-spec) utilizes multiple simultaneously active Redis servers. When Redis Cluster is used as the backplane for SignalR, all messages are delivered to all of the nodes of the cluster without code modifications to the app. However, there's a tradeoff in that increasing the number of nodes in the cluster decreases the message throughput of the backplane. This is because each message must be sent to every active node in the cluster.
```
Comment (BrennanConroy) [inline aspnetcore/signalr/redis-backplane.md:217]: Tradeoffs usually have two sides, this explains the downside, but not the upside.

Comment (guardrex) [reply aspnetcore/signalr/redis-backplane.md:217]: Yes, it's the opposite (decrease nodes = increase throughput) ... and the paragraph is calling out what happens when they're added. However, I think I can reword this to address it. *Stand-by .......*

Comment (BrennanConroy) [reply aspnetcore/signalr/redis-backplane.md:217]: No, I think the tradeoff is that you get higher availability

Comment (guardrex) [reply aspnetcore/signalr/redis-backplane.md:217]: Oh ... I misunderstood what you meant. I'll make another patch to address it as "higher availability."

Comment (guardrex) [reply aspnetcore/signalr/redis-backplane.md]: I'm not sure if I'm making this *better* or ***worse***. You might want to suggest the exact text that you'd like.

I think I detect that a separate paragraph makes sense there tho, so split it out on the last commit to see how it composes.

Comment (VilleKylmamaa) [reply aspnetcore/signalr/redis-backplane.md]: I added "achieve high availability" back in the first sentence so that the upside is clear from the start. It got dropped off in the iterations.

File: aspnetcore/signalr/redis-backplane.md [hunk 2]
```diff
@@ -212,9 +212,16 @@ services.AddSignalR()
 
 :::moniker-end
 
-## Redis Clustering
+## Redis Cluster
+
+[Redis Cluster](https://redis.io/topics/cluster-spec) utilizes multiple simultaneously active Redis servers to achieve high availability. When Redis Cluster is used as the backplane for SignalR, messages are delivered to all of the nodes of the cluster without code modifications to the app.
+
+There's a tradeoff between the number of nodes in the cluster and the availability of the backplane, increasing one decreases the other. When additional nodes are added to the cluster, the backplane's availability to process messages is reduced.
```
Comment (BrennanConroy) [inline aspnetcore/signalr/redis-backplane.md:219]: We lost the throughput messaging now 😆

Comment (guardrex) [reply aspnetcore/signalr/redis-backplane.md:219]: It seems like the main concept concern for the second paragraph is "throughput" because increasing nodes = increasing availability, which is good. The throughput of the backplane decreases when nodes are added because all nodes receive messages.

I'll try again with that 2nd paragraph 😈.

**Comment by guardrex** [comment]
Let's see how this looks ...

> There's a tradeoff between the number of nodes in the cluster and the throughput of the backplane. Increasing the number of nodes increases the availability of the cluster but decreases the throughput because messages must be transmitted to all of the nodes in the cluster.

File: aspnetcore/signalr/redis-backplane.md [hunk 3]
```diff
@@ -212,9 +212,16 @@ services.AddSignalR()
...
+
+[Redis Cluster](https://redis.io/topics/cluster-spec) utilizes multiple simultaneously active Redis servers to achieve high availability. When Redis Cluster is used as the backplane for SignalR, messages are delivered to all of the nodes of the cluster without code modifications to the app.
+
+There's a tradeoff between the number of nodes in the cluster and the throughput of the backplane. Increasing the number of nodes increases the availability of the cluster but decreases the throughput because messages must be transmitted to all of the nodes in the cluster.
+
+In the SignalR app, include all of the possible Redis nodes using either of the following approaches:
 
-[Redis Clustering](https://redis.io/topics/cluster-spec) is a method for achieving high availability by using multiple Redis servers. Clustering is supported without any code modifications to the app.
+* List the nodes in the connection string delimited with commas.
+* If using a custom behavior for connection failures, add the nodes to [`ConfigurationOptions.Endpoints`](https://stackexchange.github.io/StackExchange.Redis/Configuration#configuration-options).
```
Comment (BrennanConroy) [inline aspnetcore/signalr/redis-backplane.md:224]:
```suggestion
* If using custom behavior for connection failures, add the nodes to [`ConfigurationOptions.Endpoints`](https://stackexchange.github.io/StackExchange.Redis/Configuration#configuration-options).
```

**Review by BrennanConroy — APPROVED** [review]

**Review by guardrex — APPROVED** [review]
# 🎉

We made it! Thanks @VilleKylmamaa! 🚀
